File: src/test/java/org/dedda/bratwurst/BratwurtstTestcase.java
package org.dedda.bratwurst;

import org.dedda.bratwurst.lang.BWClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.BeforeTest;

/**
 * Created by dedda on 12/10/15.
 *
 * @author dedda
 */
public class BratwurtstTestcase {

    @BeforeMethod
    public final void __setUpBratwurstTestcase() throws Exception {
        BWClass.unregisterAll();
        this.setUp();
    }

    protected void setUp() throws Exception {}

}

File: src/test/java/org/dedda/bratwurst/BratwurtstTestcase.kt
package org.dedda.bratwurst

import org.dedda.bratwurst.lang.BWClass
import org.testng.annotations.BeforeMethod

/**
 * Created by dedda on 12/10/15.
 *
 * @author dedda
 */
open class BratwurtstTestcase {
    @BeforeMethod
    @Throws(Exception::class)
    fun __setUpBratwurstTestcase() {
        BWClass.unregisterAll()
        setUp()
    }

    @Throws(Exception::class)
    protected open fun setUp() {
    }
}